volatility3/framework/plugins/windows/registry/userassist.py

```diff
@@ -335,6 +336,20 @@ def _generator(self):
             )
             yield result
 
+    def generate_timeline(self):
+        self._reg_table_name = intermed.IntermediateSymbolTable.create(
+            self.context, self._config_path, "windows", "registry"
+        )
+
+        for row in self._generator():
+            _depth, row_data = row
+            # check the name and the timestamp to not be empty
+            if isinstance(row_data[5], str) and not isinstance(
```

Comment:
In case these indices change, *strictly* we should be checking the treegrid and interrogating it for which columns it put out and then referencing those specifically.  I assume you're going to find me an existing example that also does it this way, so I'll let it slide, but it is something we should fix across the board.  Fixing this one now would tell us a) how difficult it will be and b) act as a template for changing the others...

Reply:
Yes, this was my first time writing a generate_timeline function, and when searching across the code base - the other previous ones all used the indexes:

https://github.com/volatilityfoundation/volatility3/blob/4f69638d4d067a1fedc85f37cb62b628aa7f3256/volatility3/framework/plugins/windows/pslist.py#L291

https://github.com/volatilityfoundation/volatility3/blob/4f69638d4d067a1fedc85f37cb62b628aa7f3256/volatility3/framework/plugins/windows/sessions.py#L110

I can see how/why some name (string) based way for the column looks cleaner, but at the same time, if the indexes produced change in generator() then it should be enforced for changes in generate_timeline() of the same plugin before accepting.

Also, I am not sure how inspecting the TreeGrid would function in cases where a column name changed or was removed after generate_timeline is written. I would much prefer the index way to backtrace there since its obvious versus some vollog type print that says a column can no longer be found.

Reply:
Yeah, that's reasonable.  Ok, you convinced me.  5;)